dist/cms.js

```diff
@@ -151,12 +151,57 @@ function getWordPressContentType() {
   return content;
 }
 
+/**
+ * Obtains data about Interactivity API usage on the page.
+ *
+ * @returns {object} Object with fields `total_regions` and `total_regions_by_namespace`.
+ */
+function getInteractivityAPIUsage() {
+  // Look for data-wp-interactive regions.
+  const interactiveRegions = document.querySelectorAll('[data-wp-interactive]');
+
+  // Count the regions by namespace.
+  const regionNamespaceCounts = {};
+  interactiveRegions.forEach( region => {
+    // Extract the namespace from JSON or as a string
+    let namespace = '';
+    const regionAttribute = region.getAttribute('data-wp-interactive');
+    try {
+      const regionData = JSON.parse( regionAttribute );
+      if ( regionData.namespace && 'string' === typeof regionData.namespace ) {
+        namespace = regionData.namespace;
+      }
+    } catch ( e ) {
+      namespace = regionAttribute;
+    }
```

Comment:
@pmeenan this seems to error a lot of JSON.parse as quite often it's just a plain string rather than JSON.

Any concerns with this? Like should we test it begins with `{` before attempting to parse to reduce the number of exceptions? Or is the cost of those similar anyway?

Reply:
I can't imagine the cost of a single exception handler, even in a loop that would get run a bunch of times, would be anything more than a rounding error. No concerns here.

Reply:
This seems more legible than trying to detect conditions and maybe throw anyway so my vote would be to leave it as-is.

Reply:
Cool. Will likely be on very small strings anyway. Will merge.